lib/material/material_gui.flow

```diff
@@ -1468,4 +1478,142 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
 		\sz -> MGetFocusGroup(\parent -> TFrameT(margin, radius, fselectLift(style, \s -> mgraphicsStyle2tgraphicsStyle(parent, s)), sz)),
 		true
 	)
+}
+
+MSegmentedButtons(
+	items : [SegmentButtonItem],
+	style : [SegmentButtonsStyle],
+	graphicsStyles : [MGraphicsStyle]
+) -> Material {
+	MGetFocusGroup(\fg -> {
+		multiSelect = extractStruct(style, MMultiSelect(false)).enabled;
+		iconsOnly = extractStruct(style, MIconsOnly(false)).enabled;
+		defaultBgColor = extractStruct(graphicsStyles, MThemedFill(MSecondaryColor())).color;
+		defaultStrokeColor = extractStruct(graphicsStyles, MThemedStroke(either(fg.theme.focusOutlineColor, MAccentColor()))).color;
+		
+		paddingWithoutIcon = 12.0;
+
+		maxHeight = make(0.0);
+		lastSelected = make(None());
+
+		makeButton = \i, item -> {
+			buttonState = item.state;
+			isSelected = item.selected;
+			maxWidth = make(60.0);
+			
+			title = if (iconsOnly) "" else eitherFn(item.text, \text -> text.text, \ -> "");
+			textStyle = eitherFn(item.text, \text -> text.style, \ -> []);
+			iconName = eitherFn(item.icon, \icon -> icon.name, \ -> "");
+			iconStyle = eitherFn(item.icon, \icon -> icon.style, \ -> []);
+			accessibilityLabel = if (title != "") title else iconName;
+			buttonRole = if (multiSelect) "checkbox" else "radio";
+
+			isFirst = i == 0;
+			isLast = i == length(items) - 1;
+			createIcon = \icon -> MIcon(icon, concat([defaultBgColor], iconStyle)); 
+			createCheckIcon = MIcon("check", concat(iconStyle, [MIconSize(20.0), defaultBgColor])) |> MBorderRight(4.);
+
+			focused = make(false);
+
+			interactive = MComponent(
+				concat(
+					[
+						MOnClick(\ -> {
+							if (!multiSelect) {
+								maybeApply(getValue(lastSelected), \prev -> if (prev != isSelected) nextDistinct(prev, false));
+								if (!getValue(isSelected)) {
+									nextDistinct(lastSelected, Some(isSelected));
+								}
+							}
+							reverseBehaviour(isSelected);
+						}),
+						AccessRole(buttonRole),
+						MMobileForceHover(),
+						MButtonTitle(const(accessibilityLabel)),
+						FAccessAttribute("aria-checked", fselectLift(isSelected, b2s)),
+						MFocused(focused)
+					],
+					buttonState
+				),
+				\focus -> {
+					MSelect5(
```

Comment:
Don't use MSelect's on high levels without absolute necessity. In this case hovered and isFocused doesn't affect content inside MFrame, but will rerender it.

Next, MFixed(width, height) can easily be a dynamic TSizedWidthHeight, MFrameCustomT also support Transform as borders/cornerRadiuses.